pygmt/helpers/utils.py

```diff
@@ -145,6 +146,119 @@ def data_kind(data=None, x=None, y=None, z=None, required_z=False):
     return kind
 
 
+def non_ascii_to_octal(argstr):
+    r"""
+    Translate non-ASCII characters to their corresponding octal codes.
+
+    Currently, only characters in the ISOLatin1+ charset and
+    Symbol/ZapfDingbats fonts are supported.
+
+    Parameters
+    ----------
+    argstr : str
+        The string to be translated.
+
+    Returns
+    -------
+    str
+        The translated string.
+
+    Examples
+    --------
+    >>> non_ascii_to_octal("•‰“”±°ÿ")
+    '\\31\\214\\216\\217\\261\\260\\377'
+    >>> non_ascii_to_octal("αζΔΩ∑π∇")
+    '@~\\141@~@~\\172@~@~\\104@~@~\\127@~@~\\344@~@~\\160@~@~\\321@~'
+    >>> non_ascii_to_octal("✁❞❡➾")
+    '@%34%\\41@%%@%34%\\176@%%@%34%\\241@%%@%34%\\376@%%'
+    >>> non_ascii_to_octal("ABC ±120° DEF α ♥")
+    'ABC \\261120\\260 DEF @~\\141@~ @%34%\\252@%%'
+    """
+    # Dictionary mapping non-ASCII characters to octal codes
+    mapping = {}
+
+    # Adobe Symbol charset
+    # References:
+    # 1. https://en.wikipedia.org/wiki/Symbol_(typeface)
+    # 2. https://unicode.org/Public/MAPPINGS/VENDORS/ADOBE/symbol.txt
+    # Notes:
+    # 1. \322 and \342 are "REGISTERED SIGN SERIF" and
+    #    "REGISTERED SIGN SANS SERIF" respectively, but only "REGISTERED SIGN"
+    #    is available in the unicode table. So both are mapped to
+    #    "REGISTERED SIGN". \323, \343, \324 and \344 also have the same
+    #    problem.
+    # 2. Characters for \140, \275, \276 are incorrect.
+    mapping.update(
+        {
+            c: "@~\\" + format(i, "o") + "@~"
+            for c, i in zip(
+                " !∀#∃%&∋()∗+,−./"  # \04x-05x
+                + "0123456789:;<=>?"  # \06x-07x
+                + "≅ΑΒΧΔΕΦΓΗΙϑΚΛΜΝΟ"  # \10x-11x
+                + "ΠΘΡΣΤΥςΩΞΨΖ[∴]⊥_"  # \12x-13x
+                + "αβχδεφγηιϕκλμνο"  # \14x-15x
+                + "πθρστυϖωξψζ{|}∼"  # \16x-17x. \177 is undefined
+                + "€ϒ′≤⁄∞ƒ♣♦♥♠↔←↑→↓"  # \24x-\25x
+                + "°±″≥×∝∂•÷≠≡≈…↵"  # \26x-27x
+                + "ℵℑℜ℘⊗⊕∅∩∪⊃⊇⊄⊂⊆∈∉"  # \30x-31x
+                + "∠∇®©™∏√⋅¬∧∨⇔⇐⇑⇒⇓"  # \32x-33x
+                + "◊〈®©™∑"  # \34x-35x
+                + "〉∫⌠⌡",  # \36x-37x. \360 and \377 are undefined
```

Comment:
Symbols \140, \275 and \276 don't exist in the unicode table, so they're incorrectly shown.

Some symbols of \36x and \37x are shown as boxes in the GitHub web GUI, but they're shown correctly in Vim, so these symbols should be good.

![Screenshot from 2023-08-02 15-08-53](https://github.com/GenericMappingTools/pygmt/assets/3974108/84b17349-fced-4db4-bb20-0548c1233a8e)